src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/CoreCLR/WebRequestPSCmdlet.CoreClr.cs

```diff
@@ -174,6 +177,50 @@ internal virtual HttpClient GetHttpClient(bool handleRedirect)
                 handler.ServerCertificateCustomValidationCallback = HttpClientHandler.DangerousAcceptAnyServerCertificateValidator;
                 handler.ClientCertificateOptions = ClientCertificateOption.Manual;
             }
+            else if (CertificateValidationScript != null)
+            {
+                // validationCallBackWrapper wraps the certificateValidationClosure ScriptBlock so the async callback can set the default
+                // PowerShell Runspace in the async thread to the Runspace of the current thread. This provides the ScriptBlock access to 
+                // the current scope.
+                Runspace currentRunspace = Runspace.DefaultRunspace;
+                ScriptBlock certificateValidationClosure = CertificateValidationScript.GetNewClosure();
+                Func<HttpRequestMessage,X509Certificate2,X509Chain,SslPolicyErrors,bool> validationCallBackWrapper = 
+                    delegate(HttpRequestMessage httpRequestMessage, X509Certificate2 x509Certificate2, X509Chain x509Chain, SslPolicyErrors sslPolicyErrors)
+                    {
+                        Runspace previousRunspace = Runspace.DefaultRunspace;
+                        Runspace.DefaultRunspace = currentRunspace;
+
+                        // Set Script scope automatic variables for the closure.
+                        PSVariableIntrinsics closureVi = certificateValidationClosure.Module.SessionState.PSVariable;
+                        closureVi.Set("HttpRequestMessage", httpRequestMessage);
+                        closureVi.Set("X509Certificate2", x509Certificate2);
+                        closureVi.Set("X509Chain", x509Chain);
+                        closureVi.Set("SslPolicyErrors", sslPolicyErrors);
+                        closureVi.Set("ErrorActionPreference", "Stop");
+
+                        Boolean result = false;
+                        try
+                        {
+                            result = LanguagePrimitives.IsTrue(
+                                certificateValidationClosure.InvokeReturnAsIs(
+                                    httpRequestMessage, 
+                                    x509Certificate2, 
+                                    x509Chain, 
+                                    sslPolicyErrors
+                                )
+                            );
+                        }
+                        catch // Treat all exceptions as Certificate failures.
+                        {
+                            result = false;
+                        }
+
+                        Runspace.DefaultRunspace = previousRunspace;
```

Comment:
I'd sleep better if this was in a finally - then I don't need to worry about a change to let some exceptions through.

Reply:
I can do that.

Reply:
no longer applicable.